src/worker/builder.rs

```diff
@@ -281,6 +285,41 @@ impl<E: 'static> WorkerBuilder<E> {
         self
     }
 
+    /// Send resources consumption stats to Faktory.
+    ///
+    /// Faktory currently accepts memory stats (specifically, RSS in kilobytes)
+    /// from workers. This data is then made available in the Web UI. Call this
+    /// method on the builder in order to opt into collecting and sending stats
+    /// to the server.
+    ///
+    /// ```no_run
+    /// # tokio_test::block_on(async {
+    /// let _w = faktory::Worker::builder()
+    ///     .register_fn("jobtype", move |_| async { Ok::<(), std::io::Error>(()) })
+    ///     .with_sysinfo()
+    ///     .connect()
+    ///     .await
+    ///     .unwrap();
+    /// # });
+    /// ```
+    ///
+    /// Note that enabling this behavior might imply that the running program
+    /// may need some extra capabilities, e.g. - for Linix platform - permissions
+    /// to read `/proc/[PID]/statm` file, where PID is your running program's id.
+    /// No system-wide or other processes data gets collected. Furthermore, it is
+    /// only the minimal amount of data (RSS as of Faktory v1.9.3) about the current
+    /// process that gets scraped and sent over to the Faktory server periodically.
+    ///
+    /// Also note that this does nothing in case the target OS is not supported by the
+    /// [`sysinfo`](https://docs.rs/sysinfo/latest/sysinfo/index.html#supported-oses)
+    /// crate which is being used internally.
```

Comment:
Should we maybe return a `bool` so that there's at least _some_ indication whether this worked or not?